langchain/src/agents/tools/base.ts

```diff
@@ -19,21 +24,27 @@ export abstract class Tool {
 
   protected abstract _call(arg: string): Promise<string>;
 
-  async call(arg: string, verbose?: boolean): Promise<string> {
+  async call(
+    arg: string,
+    callerId?: RunId,
+    verbose?: boolean
+  ): Promise<string> {
     const _verbose = verbose ?? this.verbose;
-    await this.callbackManager.handleToolStart(
+    const values = await this.callbackManager.handleToolStart(
       { name: this.name },
       arg,
+      callerId,
       _verbose
     );
     let result;
+    const runId = values[TRACER_RUN_ID];
```

Comment:
I don't think this is a good idea, we're leaking an implementation detail of the tracer (one of many callback handlers) into the whole codebase. This is why I had the callback manager be the one that assigns runIds. What's the advantage of this?